### PR TITLE
feat: add react native config

### PR DESCRIPTION
**Pull request by ASteinheiser**

### CODE REVIEW

File: packages/eslint-config-react-native/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  plugins: ['react-native', 'react', 'jsx-a11y', 'import', '@typescript-eslint'],
```
Comment (w33ble) [inline packages/eslint-config-react-native/index.js]: Is there a reason you're not using any of the [react-native supported rules](https://github.com/intellicode/eslint-plugin-react-native#list-of-supported-rules)?

File: packages/eslint-config-react-native/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  rules: {
```
Comment (w33ble) [inline packages/eslint-config-react-native/index.js]: All of the rules you're disabling here seem like project-specific overrides. Is there something I'm missing, or can you make a case for these things?

File: packages/eslint-config-react/index.js [hunk 3]
```diff
@@ -7,14 +7,18 @@ module.exports = {
     },
   },
   rules: {
+    'no-use-before-define': 'off',
+    'lines-between-class-members': 'off',
+    'class-methods-use-this': 'off',
```
Comment (w33ble) [inline packages/eslint-config-react/index.js]: Why are you disabling react rules here? These seem like project-specific overrides. Some of them can be auto-fix too I believe.

File: packages/eslint-config-react/index.js [hunk 4]
```diff
@@ -7,14 +7,18 @@ module.exports = {
...
     // react
+    'react/prop-types': ['warn', { skipUndeclared: true }],
```
Comment (w33ble) [inline packages/eslint-config-react/index.js]: This is project-specific override, please remove it.

File: packages/eslint-config-react/index.js [hunk 5]
```diff
@@ -7,14 +7,18 @@ module.exports = {
...
     'react-hooks/exhaustive-deps': 'warn',
     'react-hooks/rules-of-hooks': 'error',
     'react/destructuring-assignment': 'off',
     'react/display-name': 'error',
     'react/forbid-prop-types': 'error',
     'react/jsx-closing-tag-location': 'off', // defer to prettier
-    'react/jsx-filename-extension': ['error', { extensions: ['.jsx', '.tsx'] }],
+    'react/jsx-filename-extension': ['error', { extensions: ['.js', '.jsx', '.tsx'] }],
```
Comment (w33ble) [inline packages/eslint-config-react/index.js]: This is project-specific override, please remove it. Files with jsx in them should generally only have jsx or tsx extensions.

File: packages/eslint-config-typescript/index.js [hunk 6]
```diff
@@ -6,6 +6,8 @@ module.exports = {
       extends: ['plugin:import/typescript', 'plugin:@typescript-eslint/recommended'],
       rules: {
         'no-unused-vars': 'off',
+        '@typescript-eslint/no-use-before-define': 'off',
+        '@typescript-eslint/camelcase': 'off',
```
Comment (w33ble) [inline packages/eslint-config-typescript/index.js]: These are project-specific overrides, please remove these.

File: packages/eslint-config/index.js [hunk 7]
```diff
@@ -16,8 +16,12 @@ module.exports = {
   extends: ['airbnb-base', 'prettier'],
   plugins: ['prettier', 'import'],
   rules: {
+    'max-lines': ['warn', { max: 800 }],
     strict: 'off',
     'no-debugger': 'error',
+    'no-underscore-dangle': 'off',
+    'no-plusplus': 'off',
+    'no-bitwise': 'off',
```
Comment (w33ble) [inline packages/eslint-config/index.js]: These are project-specific overrides, please remove these.

File: packages/eslint-config/index.js [hunk 8]
```diff
@@ -34,6 +38,7 @@ module.exports = {
       'error',
       {
         groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
+        'newlines-between': 'always',
```
Comment (w33ble) [inline packages/eslint-config/index.js]: This is a project-specific override, please remove it.

I personally really dislike all the newlines in imports, but you can try to make the case for this at a later time if you'd like.

File: packages/eslint-config-react-native/index.js (hunk 1 above)
Comment (ASteinheiser) [reply packages/eslint-config-react-native/index.js]: Gotcha, I just added all their rules

File: packages/eslint-config-react-native/index.js (hunk 2 above)
Comment (ASteinheiser) [reply packages/eslint-config-react-native/index.js]: Good point, I moved those to my projects override

**Pull request merged**